elastic/eui
# [Docs] Fixes all demos importing `@elastic/charts` to work correctly

## Summary

closes https://github.com/elastic/eui/issues/6135

The above PR only mentions `EuiMarkdownEditor`, but as I was working on this I realized that it fixed many issues for our Charts documentation pages as well and enables several of our more standalone Demo JS snippets to now work correctly in CodeSandbox.

This PR also fixes several charts examples not actually correctly working in dark mode - all examples should now correctly dynamically set light vs dark theme.

Code changes contain a lot of cleanup so I recommend QAing staging vs prod instead if comparing/checking for regressions

## QA

- Go to http://localhost:8030/#/editors-syntax/markdown-plugins#putting-it-all-together-a-simple-chart-plugin
    - [ ] Switch to dark mode and confirm the chart on the dev page now correctly uses dark colors (check by hovering over a bar) [compared to prod](https://elastic.github.io/eui/#/editors-syntax/markdown-plugins#putting-it-all-together-a-simple-chart-plugin)
    - [ ] Open the Demo JS tab, click the Codesandbox link, and confirm everything loads & renders without error
- Go to http://localhost:8030/#/elastic-charts/creating-charts
    - [ ] Open the Demo JS tab, click the Codesandbox link, and confirm everything loads & renders without error in both light & dark mode

### General checklist


~- [ ] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**~
~- [ ] Checked in **mobile**~
~- [ ] Props have proper **autodocs** and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md#adding-playground-toggles)**~
~- [ ] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md)**~
~- [ ] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/cypress-testing.md) tests**~
~- [ ] Checked for **breaking changes** and labeled appropriately~
~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
~- [ ] Updated the **[Figma](https://www.figma.com/community/file/964536385682658129)** library counterpart~
~- [ ] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md#changelog)** entry exists and is marked appropriately~
